manifests/charts/istio-control/istio-discovery/templates/istiod-injector-configmap.yaml

```diff
@@ -48,6 +48,10 @@ data:
       sidecar: |
 {{ .Files.Get "files/injection-template.yaml" | trim | indent 8 }}
 {{- end }}
+{{- if not (hasKey .Values.sidecarInjectorWebhook.templates "gateway") }}
```

Comment:
Why do we keep the gateway template in values ?

Reply:
This `if` is just so that if a user defines their own `"gateway"` template, via the existing `.Values.sidecarInjectorWebhook.templates` option, we use theirs instead of overwriting it with the builtin.

Reply:
I mean: do we need the extra if ?